### PR TITLE
Add polygon data container

### PR DESCRIPTION
**Pull request by brittonsmith**
This is a template of the classes required to implement a polygon data container.

Some useful documentation [here](https://yt-project.org/docs/dev/developing/creating_datatypes.html).

### CODE REVIEW

**Comment by arevill2** [comment]
Update to polygon function:
- Support shapefile multi-feature polygons 
- Generate binary masks as numpy array for polygon selection

File: yt_geotiff/tsearch.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
```
Comment (brittonsmith) [inline yt_geotiff/tsearch.h:1]: Just to be clear, this file serves no other purpose than I get a compile error if it is not here.

**Comment by brittonsmith** [comment]
The latest changes will resolve Issue #37.

File: README.md [hunk 2]
```diff
@@ -262,3 +269,35 @@ bounding box enclosing the container.
...
+>>> landsat_data = "C:/path/to/Landsat_test_data/LC08_L2SP_171060_20210227_20210304_02_T1_SR_B1.TIF"
+>>> ds = yt.load(landsat_data
+yt : [INFO     ] 2021-05-31 12:58:36,740 Parameters: domain_dimensions         = [7581 7741    1]
+yt : [INFO     ] 2021-05-31 12:58:36,743 Parameters: domain_left_edge          = [ 361485. -116415.       0.] m
+yt : [INFO     ] 2021-05-31 12:58:36,744 Parameters: domain_right_edge         = [5.88915e+05 1.15815e+05 1.00000e+00] m
+```
+
+Load and read a polygon shapefile into yt_geotiff
+```
+>>> multifeature_shapefile = "C:/path/to/esri_shapefile/multi_feature_polygon.shp"
```
Comment (brittonsmith) [inline README.md:292]:
```suggestion
>>> multi_shapefile = "C:/path/to/esri_shapefile/multi_feature_polygon.shp"
```

**Pull request merged**